astropy/coordinates/abstract.py

```diff
@@ -0,0 +1,1550 @@
+            # We don't want to fail for this case.
+            differential_type = r.DIFFERENTIAL_CLASSES.get(base.get_name(), None)
+
+        elif differential_type in r.DIFFERENTIAL_CLASSES:
+            differential_type = r.DIFFERENTIAL_CLASSES[differential_type]
+
+        elif differential_type is not None and (
+            not isinstance(differential_type, type)
+            or not issubclass(differential_type, r.BaseDifferential)
+        ):
+            raise ValueError(
+                "Differential is {differential_type!r} but must be a BaseDifferential"
+                f" class or one of the string aliases {list(r.DIFFERENTIAL_CLASSES)}"
+            )
+        repr_classes[name] = differential_type
+    return repr_classes
+
+class AbstractCoordinate(MaskableShapedLikeNDArray):
+    # TODO APE: docstring 
+
+    default_representation = None
+    default_differential = None
+
+    # Specifies special names and units for representation and differential
+    # attributes.
+    frame_specific_representation_info = {}
+
+    def __init_subclass__(cls, **kwargs):
+        # We first check for explicitly set values for these:
+        default_repr = getattr(cls, "default_representation", None)
+        default_diff = getattr(cls, "default_differential", None)
+        repr_info = getattr(cls, "frame_specific_representation_info", None)
+        # Then, to make sure this works for subclasses-of-subclasses, we also
+        # have to check for cases where the attribute names have already been
+        # replaced by underscore-prefaced equivalents by the logic below:
+        if default_repr is None or isinstance(default_repr, property):
+            default_repr = getattr(cls, "_default_representation", None)
+
+        if default_diff is None or isinstance(default_diff, property):
+            default_diff = getattr(cls, "_default_differential", None)
+
+        if repr_info is None or isinstance(repr_info, property):
+            repr_info = getattr(cls, "_frame_specific_representation_info", None)
+
+        repr_info = cls._infer_repr_info(repr_info)
+
+        # Make read-only properties for the frame class attributes that should
+        # be read-only to make them immutable after creation.
+        # We copy attributes instead of linking to make sure there's no
+        # accidental cross-talk between classes
+        cls._create_readonly_property(
+            "default_representation",
+            default_repr,
+            "Default representation for position data",
+        )
+        cls._create_readonly_property(
+            "default_differential",
+            default_diff,
+            "Default representation for differential data (e.g., velocity)",
+        )
+        cls._create_readonly_property(
+            "frame_specific_representation_info",
+            copy.deepcopy(repr_info),
+            "Mapping for frame-specific component names",
+        )
+
+        # TODO APE: keep?
```

Comment:
@adrn @mhvk @nstarman - do we need this?

Reply:
I would say not in `AbstractCoordinate`, but yes in `SkyCoord`.